src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4359,7 +4359,7 @@ static QVariant fcnProject( const QVariantList &values, const QgsExpressionConte
 {
   QgsGeometry geom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
 
-  if ( geom.type() != QgsWkbTypes::PointGeometry )
+  if ( QgsWkbTypes::flatType( geom.wkbType() ) != QgsWkbTypes::Point )
```

Comment:
Would be nice here to check it it's a multipart geometry with a single part only first. You can use QgsAbstractGeometry.simplifiedTypeRef to make this easy

Reply:
But that would require patching https://github.com/qgis/QGIS/blob/master/src/core/geometry/qgspoint.cpp#L759 to avoid hitting https://github.com/qgis/QGIS/blob/master/src/core/geometry/qgspoint.cpp#L49

Do you think it qualifies as a bugfix?

Reply:
@nyalldawson nevermind, I've found a way that doesn't require messing with QgsPoint